### ISSUE TITLE
Exceptions while using model.fit

### ISSUE THREAD

**Issue by XpRienzo** [plaidml/plaidml#95]
I was training a LSTM on 0.3.0rc1 when this error occured
```
---------------------------------------------------------------------------
Unknown                                   Traceback (most recent call last)
~\Anaconda3\lib\site-packages\plaidml\tile.py in bind(self, bindings)
    253             try:
--> 254                 outputs[output_name] = applier.add_output(output_name)
    255             except BaseException as e:

~\Anaconda3\lib\site-packages\plaidml\__init__.py in add_output(self, name)
   1367     def add_output(self, name):
-> 1368         return Var(_lib().plaidml_apply_alloc_output(self, name.encode()))
   1369 

~\Anaconda3\lib\site-packages\plaidml\__init__.py in _check_err(self, result, func, args)
    717             return None
--> 718         self.raise_last_status()
    719 

~\Anaconda3\lib\site-packages\plaidml\library.py in raise_last_status(self)
    130     def raise_last_status(self):
--> 131         raise self.last_status()
    132 

Unknown: b'Missing output binding "N"'

During handling of the above exception, another exception occurred:

AttributeError                            Traceback (most recent call last)
<ipython-input-19-c50a2dc352a6> in <module>()
----> 1 model.fit(Xtrain, Ytrain, epochs = 100)

~\Anaconda3\lib\site-packages\keras\engine\training.py in fit(self, x, y, batch_size, epochs, verbose, callbacks, validation_split, validation_data, shuffle, class_weight, sample_weight, initial_epoch, steps_per_epoch, validation_steps, **kwargs)
   1573         else:
   1574             ins = x + y + sample_weights
-> 1575         self._make_train_function()
   1576         f = self.train_function
   1577 

~\Anaconda3\lib\site-packages\keras\engine\training.py in _make_train_function(self)
    965                                                  updates=updates,
    966                                                  name='train_function',
--> 967                                                  **self._function_kwargs)
    968 
    969     def _make_test_function(self):

~\Anaconda3\lib\site-packages\plaidml\keras\backend.py in function(inputs, outputs, updates, name)
    718     if name == None:
    719         name = ''
--> 720     return _Function(inputs, outputs, updates, name)
    721 
    722 

~\Anaconda3\lib\site-packages\plaidml\keras\backend.py in __init__(self, inputs, outputs, updates, name)
    121                                    _device(),
    122                                    list(zip(self._input_names, inputs)),
--> 123                                    list(zip(self._output_names, outputs)), updates)
    124         self._invoker = plaidml.Invoker(_ctx, self._func)
    125 

~\Anaconda3\lib\site-packages\plaidml\tile.py in compose(ctx, dev, inputs, outputs, updates)
   1003             to_be_bound.extend(reqs)
   1004             continue
-> 1005         bindings.insert(op, op.bind(bindings))
   1006         updates.extend(op.side_effects)
   1007 

~\Anaconda3\lib\site-packages\plaidml\tile.py in bind(self, bindings)
    255             except BaseException as e:
    256                 raise Exception('Failed to add output \'{}\' in op {}: {}; code={}'.format(
--> 257                     output_name, self.name, e.message, self.code))
    258         return outputs
    259 

AttributeError: 'Unknown' object has no attribute 'message'
```

**Comment by XpRienzo** [comment]
Now facing this issue while training

```Epoch 1/100
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-20-c50a2dc352a6> in <module>()
----> 1 model.fit(Xtrain, Ytrain, epochs = 100)

~\Anaconda3\lib\site-packages\keras\engine\training.py in fit(self, x, y, batch_size, epochs, verbose, callbacks, validation_split, validation_data, shuffle, class_weight, sample_weight, initial_epoch, steps_per_epoch, validation_steps, **kwargs)
   1596                               initial_epoch=initial_epoch,
   1597                               steps_per_epoch=steps_per_epoch,
-> 1598                               validation_steps=validation_steps)
   1599 
   1600     def evaluate(self, x, y,

~\Anaconda3\lib\site-packages\keras\engine\training.py in _fit_loop(self, f, ins, out_labels, batch_size, epochs, verbose, callbacks, val_f, val_ins, shuffle, callback_metrics, initial_epoch, steps_per_epoch, validation_steps)
   1181                     batch_logs['size'] = len(batch_ids)
   1182                     callbacks.on_batch_begin(batch_index, batch_logs)
-> 1183                     outs = f(ins_batch)
   1184                     if not isinstance(outs, list):
   1185                         outs = [outs]

~\Anaconda3\lib\site-packages\plaidml\keras\backend.py in __call__(self, inputs)
    138                 val = plaidml.Real(val)
    139             else:
--> 140                 val = variable(val, dtype=self._input_types[name]).var
    141             self._invoker.set_input(name, val)
    142 

~\Anaconda3\lib\site-packages\plaidml\keras\backend.py in variable(value, dtype, name, constraint)
   1681                                               *value.shape))
   1682         with tensor.mmap_discard(_ctx) as view:
-> 1683             view.copy_from_ndarray(value)
   1684             view.writeback()
   1685         return ptile.Value.from_var(tensor, value.shape, ptile.NUMPY_DTYPE_TO_PLAIDML[dtype],

~\Anaconda3\lib\site-packages\plaidml\__init__.py in copy_from_ndarray(self, src)
   1188             src = src.view(dtype='uint16')
   1189         dst = np.ctypeslib.as_array(self, shape=src.shape)
-> 1190         np.copyto(dst, src)
   1191 
   1192     def copy_to_ndarray(self, dst):

TypeError: Cannot cast scalar from dtype('float64') to dtype('int32') according to the rule 'same_kind'```

**Comment by XpRienzo** [comment]
Other than this, when I removed the expected input type from input layer, then it did run, apart from some warnings
```INFO:plaidml:b'Analyzing Ops: 6200 of 8730 operations complete'
WARNING:plaidml:b'Reached scheduler optimization timeout'```
The training was slow, but I guess that's because of sparse tensors not being implemented as mentioned in #91

**Comment by jbruestle** [comment]
Closing since this is mostly a dup of other issues at this point.  The typing thing possible warrants a new ticket, but @XpRienzo, I'll let you open a more specific one if it's still an issue for you.
